syntax = "proto3";

package dating.service.profiles;


service ProfilesService {
    rpc Create(ProfileCreateRequest) returns (ProfileCreateResponse);
    rpc Get(ProfilesGetRequest) returns (ProfilesGetResponse);
    rpc Update(ProfileUpdateRequest) returns (ProfilesUpdateResponse);
    rpc Delete(ProfileDeleteRequest) returns (ProfilesDeleteResponse);
}


message ProfileCreateRequest {
    string account_id = 1;
    string first_name = 2;
    string last_name = 3;
    int32 age = 4;
    Gender gender = 5;
    optional string biography = 6;
    string language_locale = 7;
    repeated string image_base64_list = 8;
    float lat = 9;
    float lon = 10;
}


message ProfilesGetRequest {
    oneof identifier {
        string id = 1;
        string account_id = 2;
    }
}


message ProfileUpdateRequest {
    message UpdateData {
        optional string account_id = 1;
        optional string first_name = 2;
        optional string last_name = 3;
        optional int32 age = 4;
        optional Gender gender = 5;
        optional string biography = 6;
        optional string language_locale = 7;
        repeated string image_base64_list = 8;
        optional float lat = 9;
        optional float lon = 10;
    }
    string id = 1;
    UpdateData data = 2;
}


message ProfileDeleteRequest {
    string id = 1;
}


message ProfileCreateResponse {
    string id = 1;
    string account_id = 2;
    string first_name = 3;
    string last_name = 4;
    int32 age = 5;
    Gender gender = 6;
    string biography = 7;
    string language_locale = 8;
    string created_at = 9;
    string updated_at = 10;
    repeated string image_base64_list = 11;
    float lat = 12;
    float lon = 13;
}


message ProfilesGetResponse {
    string id = 1;
    string account_id = 2;
    string first_name = 3;
    string last_name = 4;
    int32 age = 5;
    Gender gender = 6;
    string biography = 7;
    string language_locale = 8;
    string created_at = 9;
    string updated_at = 10;
    repeated string image_base64_list = 11;
    float lat = 12;
    float lon = 13;
}


message ProfilesUpdateResponse {
    string result = 1;
}


message ProfilesDeleteResponse {
    string result = 1;
}


enum Gender {
    GENDER_DEFAULT = 0;
    GENDER_MALE = 1;
    GENDER_FEMALE = 2;
}
